conn.go

```diff
@@ -900,7 +900,12 @@ func (c *Conn) queueRequest(opcode int32, req interface{}, res interface{}, recv
 	switch opcode {
 	case opClose:
 		// always attempt to send close ops.
-		c.sendChan <- rq
+		select {
+		case c.sendChan <- rq:
+		case <-time.After(time.Second):
+			c.logger.Printf("gave up trying to send opClose to server")
+			rq.recvChan <- response{-1, ErrConnectionClosed}
```

Comment:
This also blocks and is another state in the request path, i am worried about adding this without a test case showing the issue plus validation of a fix.

Reply:
The recvChan is created right in this function, buffered. Shall not block.